fairlearn/metrics/_extra_metrics.py

```diff
@@ -6,38 +6,214 @@
 These are metrics which are not part of `scikit-learn`.
 """
 
+import numpy as np
 import sklearn.metrics as skm
 
 from ._balanced_root_mean_squared_error import _balanced_root_mean_squared_error  # noqa: F401
 from ._mean_predictions import mean_prediction, _mean_overprediction, _mean_underprediction  # noqa: F401,E501
 from ._selection_rate import selection_rate  # noqa: F401,E501
 
+_TOO_MANY_UNIQUE_Y_VALS = "Must have no more than two unique y values"
+_RESTRICTED_VALS_IF_POS_LABEL_NONE = "If pos_label is not specified, values must be take from {0, 1} or {-1, 1}"  # noqa: E501
+_NEED_POS_LABEL_IN_Y_VALS = "Must have pos_label in y values"
+_NO_SUCH_ELEMENT = "16763842-875f-467c-9e58-24987dfe5d7d"
 
-def true_positive_rate(y_true, y_pred, sample_weight=None):
-    r"""Calculate the true positive rate (also called sensitivity, recall, or hit rate)."""
-    tnr, fpr, fnr, tpr = skm.confusion_matrix(
-        y_true, y_pred, sample_weight=sample_weight, labels=[0, 1], normalize="true").ravel()
+
+def _get_labels_for_confusion_matrix(unique_labels, pos_label):
+    r"""Figure out the labels argument for skm.confusion_matrix.
+
+    This is an internal method used by the true/false positive/negative
+    rate metrics (and hence are restricted to binary data). We compute
+    these using the confusion matrix.
+    This method prepares the `labels` argument of
+    :py:func:`sklearn.metrics.confusion_matrix` based on the
+    user's specifications.
+
+    Parameters
+    ----------
+    unique_labels : array-like
+        The sorted list of unique labels in the y arrays
+
+    pos_label : scalar
+        The value in the true and predicted arrays to treat as positive.
+        If this is not set, then the unique_labels must be a subset of
+        {0, 1} or {-1, 1}, and it will then be set to 1
+
+    Returns
+    -------
+    list
+        A two element list, consisting of the unique labels
+        with the positive label listed last. This array will
+        always be two elements, even if the unique_labels array
+        only has one element.
+    """
+    if len(unique_labels) > 2:
+        raise ValueError(_TOO_MANY_UNIQUE_Y_VALS)
+
+    # Set pos_label if needed
+    if pos_label is None:
+        labels01 = frozenset([0, 1])
+        labels11 = frozenset([-1, 1])
+        if labels01.issuperset(unique_labels) or labels11.issuperset(unique_labels):
+            pos_label = 1
+        else:
+            raise ValueError(_RESTRICTED_VALS_IF_POS_LABEL_NONE)
+
+    # Ensure unique_labels has two elements
+    if len(unique_labels) == 1:
+        if unique_labels[0] == pos_label:
+            unique_labels = [_NO_SUCH_ELEMENT, pos_label]
```

Comment:
if pos_label is numeric then i think this will fail:
```
>>> skm.confusion_matrix([1,1,1,1],[1,1,1,1], labels=['not 1',1])
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/mnt/c/Users/mdudik/projects/fairlearn/.venv/lib/python3.6/site-packages/sklearn/metrics/_classification.py", line 277, in confusion_matrix
    raise ValueError("At least one label specified must be in y_true")
ValueError: At least one label specified must be in y_true
>>>
```

Reply:
oh... i see why this doesn't trigger an error below--because you have unique_labels and cm_labels.

Reply:
Yes, that exception looks rather familiar from when I was developing this.